SentinelOne/sentinelone_module/logs/connector.py

```diff
@@ -0,0 +1,123 @@
+from functools import cached_property
+import time
+
+from datetime import datetime, timedelta, timezone
+from prometheus_client import Counter, Histogram
+from management.mgmtsdk_v2.exceptions import UnauthorizedException
+from management.mgmtsdk_v2.services.activity import ActivitiesFilter
+from management.mgmtsdk_v2.services.threat import ThreatQueryFilter
+from management.mgmtsdk_v2_1.mgmt import Management
+from sekoia_automation.connector import Connector
+
+from sentinelone_module.logs.configuration import SentinelOneLogsConnectorConfiguration
+
+SENTINEL_ONE_HOW_MANY_DAYS_OF_EXISTING_LOGS_TO_RETRIEVE = 1
+
+# Declare prometheus metrics
+prom_namespace = "sicconfapi_intakes"
+
+OUTCOMING_EVENTS = Counter(
+    name="forwarded_events",
+    documentation="Number of events forwarded to SEKOIA.IO",
+    namespace=prom_namespace,
+    labelnames=["datasource", "intake_key"],
+)
+
+FORWARD_EVENTS_DURATION = Histogram(
+    name="forward_events_duration",
+    documentation="Duration to collect and forward events",
+    namespace=prom_namespace,
+    labelnames=["datasource", "intake_key"],
+)
+
+
+class SentinelOneLogsConnector(Connector):
+    configuration: SentinelOneLogsConnectorConfiguration
+
+    @cached_property
+    def management_client(self):
+        return Management(hostname=self.configuration.management_domain, api_token=self.configuration.api_key)
+
+    @property
+    def logs_retrieval_start_date(self) -> str:
+        return (
+            datetime.now(tz=timezone.utc) - timedelta(days=SENTINEL_ONE_HOW_MANY_DAYS_OF_EXISTING_LOGS_TO_RETRIEVE)
+        ).isoformat()
+
+    def validate_settings(self):
+        if not self.configuration.api_key:
+            self.log(f"Connector {self.configuration.uuid} missing api_key value", level="warning")
+            return
+
+        if not self.configuration.management_domain:
+            self.log(f"Connector {self.configuration.uuid} missing management_domain value", level="warning")
+            return
+
+        try:
+            self.management_client.system.get_info()
+            self.configuration.validated = True
```

Comment:
I'm not sure that it's useful in the connector. I think that the validation was done one time in the previous integration and this boolean was saved in the database. Here, It's preferable to always check the setting at the starting of the connector.

Reply:
Ok so no need to have a `configuration.validated` field, just calling `get_info()` at the start of the run loop is enough?